core/src/no/group15/playmagic/PlayMagic.kt

```diff
@@ -1,28 +1,24 @@
 package no.group15.playmagic
 
 import com.badlogic.gdx.*
+import com.badlogic.gdx.assets.AssetManager
 import com.badlogic.gdx.graphics.GL20
 import com.badlogic.gdx.graphics.g2d.SpriteBatch
+import ktx.async.KtxAsync
+import ktx.inject.*
 import no.group15.playmagic.ui.AppState
 import no.group15.playmagic.ui.views.MainMenuView
 
 
-class PlayMagic : ApplicationListener {
+class PlayMagic(private val logLevel: Int) : ApplicationListener {
 
-	private lateinit var batch: SpriteBatch
-
-	var appState: AppState? = null
-		set(value) {
-			appState?.dispose()
-			field = value
-			appState?.create()
-			appState?.resize(Gdx.graphics.width, Gdx.graphics.height)
-		}
+	private lateinit var appState: AppState
+	private lateinit var injectContext: Context
 
 	private val commonInput = object : InputAdapter() {
 		override fun keyUp(keycode: Int): Boolean {
 			if (keycode == Input.Keys.ESCAPE || keycode == Input.Keys.BACK) {
```

Comment:
Can be rewritten to `if (keycode in setOf(Input.Keys.ESCAPE, Input.Keys.BACK) ...`

Reply:
👍